pkg/systemd/unit.go

```diff
@@ -154,61 +173,120 @@ func (m *UnitManager) EnsureUnits(ctx context.Context, nc *scyllav1alpha1.NodeCo
 		return unit.FileName
-	}, requiredUnits...)
+	})
 
 	err = m.WriteStatus(status)
 	if err != nil {
-		return fmt.Errorf("can't write status: %w", err)
+		return progressingMessages, fmt.Errorf("can't write status: %w", err)
 	}
 
-	for _, requiredUnit := range requiredUnits {
-		klog.V(4).InfoS("Ensuring unit", "Name", requiredUnit.FileName)
-		requiredUnitPath := m.GetUnitPath(requiredUnit.FileName)
+	for _, managedRequiredUnit := range managedRequiredUnits {
+		klog.V(4).InfoS("Ensuring unit", "Name", managedRequiredUnit.FileName)
+		managedRequiredUnitPath := m.GetUnitPath(managedRequiredUnit.FileName)
 
 		exists := true
-		_, err := os.Stat(requiredUnitPath)
+		_, err = os.Stat(managedRequiredUnitPath)
 		if err != nil {
 			if os.IsNotExist(err) {
 				exists = false
 			} else {
-				return fmt.Errorf("can't stat unit %q: %w", requiredUnitPath, err)
+				return progressingMessages, fmt.Errorf("can't stat unit %q: %w", managedRequiredUnitPath, err)
 			}
 		}
 
-		err = os.WriteFile(requiredUnitPath, requiredUnit.Data, 0666)
+		err = os.WriteFile(managedRequiredUnitPath, managedRequiredUnit.Data, 0666)
 		if err != nil {
-			return fmt.Errorf("can't write unit %q: %w", requiredUnitPath, err)
+			return progressingMessages, fmt.Errorf("can't write unit %q: %w", managedRequiredUnitPath, err)
 		}
 
 		if !exists {
-			klog.V(2).InfoS("Mount unit has been created", "Name", requiredUnit.FileName)
+			klog.V(2).InfoS("Mount unit has been created", "Name", managedRequiredUnit.FileName)
 			recorder.Eventf(
 				nc,
 				corev1.EventTypeNormal,
 				"MountCreated",
 				"Mount unit %s has been created",
-				requiredUnit.FileName,
+				managedRequiredUnit.FileName,
 			)
 		}
 	}
 
 	// Reload unit definitions to enable and start them.
 	err = control.DaemonReload(ctx)
 	if err != nil {
-		return fmt.Errorf("can't reload systemd: %w", err)
+		return progressingMessages, fmt.Errorf("can't reload systemd: %w", err)
 	}
 
-	for _, requiredUnit := range requiredUnits {
-		klog.V(2).InfoS("Enabling and starting unit", "Name", requiredUnit.FileName)
-		err = control.EnableAndStartUnit(ctx, requiredUnit.FileName)
+	for _, managedRequiredUnit := range managedRequiredUnits {
+		klog.V(2).InfoS("Enabling unit", "Name", managedRequiredUnit.FileName)
+		err = control.EnableUnit(ctx, managedRequiredUnit.FileName)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("can't enable unit %q: %w", managedRequiredUnit.FileName, err))
+			continue
+		}
+
+		unitStatuses, err = control.GetUnitStatuses(ctx, []string{managedRequiredUnit.FileName})
+		if err != nil {
+			errs = append(errs, fmt.Errorf("can't get unit statuses for unit %q: %w", managedRequiredUnit.FileName, err))
+			continue
+		}
+		if len(unitStatuses) == 0 {
+			errs = append(errs, fmt.Errorf("missing status for unit %q", managedRequiredUnit.FileName))
+			continue
+		}
+		unitStatus := unitStatuses[0]
+
+		switch unitStatus.ActiveState {
+		case activeStateActive:
+
```

Comment:
Adding a comment here could explain nothing is expected to do here. Without it may suggest author forgot to implement this branch.

Reply:
or a log